### PR TITLE
Add failing test for namespaced keys

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref #22

### CODE REVIEW

File: flatfs.go [hunk 1]
```diff
@@ -116,8 +116,11 @@ func (fs *Datastore) ShardStr() string {
 }
 
 func (fs *Datastore) encode(key datastore.Key) (dir, file string) {
+	fmt.Printf("key: %s\n", key.String())
```
Comment (Stebalien) [inline flatfs.go]: These look like debugging print statements.

File: flatfs_test.go [hunk 2]
```diff
@@ -71,6 +71,39 @@ func testPut(dirFunc mkShardFunc, t *testing.T) {
...
+	if !ok {
+		t.Fatalf("expected []byte from Get, got %T: %v", data, data)
+	}
+
+	if g, e := string(buf), input; g != e {
+		t.Fatalf("Get gave wrong content: %q != %q", g, e)
+	}
+}
+
+func TestPutGetNamespaces(t *testing.T) { tryAllShardFuncs(t, testPutGetNamespaces) }
```
Comment (Stebalien) [inline flatfs_test.go]: One line functions are generally not idiomatic go. Try running `go fix ./...` and/or `go fmt ./...` on your repo to fix stuff like this.

File: flatfs.go (hunk 1 above)
Comment (dignifiedquire) [reply flatfs.go]: Yep they are, until we figured out how to fix the issue.

File: flatfs_test.go (hunk 2 above)
Comment (dignifiedquire) [reply flatfs_test.go]: I just copy and pasted from the above test, to keep the original style.

Comment (dignifiedquire) [reply flatfs_test.go]: formatting cleanup should be probably done in a different commit/pr

**Pull request closed without merging**